recipes/xapian-core/all/conanfile.py

```diff
@@ -45,13 +47,12 @@ def requirements(self):
         self.requires("zlib/1.2.11")
 
     def build_requirements(self):
-        if tools.os_info.is_windows and not tools.get_env("CONAN_BASH_PATH") and \
-                tools.os_info.detect_windows_subsystem() != "msys2":
-            self.build_requires("msys2/20190524")
+        if tools.os_info.is_windows and not tools.get_env("CONAN_BASH_PATH"):
+            self.build_requires("msys2/20200517")
```

Comment:
I guess it's time to make the switch since 20200517 is not available anymore in this git repo.
```suggestion
            self.build_requires("msys2/cci.latest")
```

Reply:
Can't those people then simply add:
```
[build_requires]
xapian-core:msys2/20200517
```
to their profile?

Reply:
they could I guess, not even sure it works, but it's not what I call "build a recipe out of the box". Why consumers should dig into internal details like the underlying build system of a recipe?

Reply:
Indeed, some syntax to override package versions would be nice.
e.g.
```
[build_requires]
msys2/20200517:override
```
Which would NOT add the msys2 recipe, but would if a msys2 package is required, switch it with `msys2/20200517`.
I don't know whether conan has this capability.

Reply:
My concern is not about host or target, but build machine. 2 profiles won't help when a recipe has msys2 in build requirements.